omnigibson/examples/objects/import_custom_object.py

```diff
@@ -43,108 +45,112 @@
 
     # Sanity check mesh type
-    valid_formats = trimesh.available_formats()
     mesh_format = asset_path.split(".")[-1]
 
     # If we're not a URDF, import the mesh directly first
+    urdf_dep_paths = None
+    temp_dirs = []
     if mesh_format != "urdf":
-        assert_valid_key(key=mesh_format, valid_keys=valid_formats, name="mesh format")
-
-        # Load the mesh
-        visual_mesh: trimesh.Trimesh = trimesh.load(asset_path, force="mesh", process=False)
-
-        # Generate collision meshes if requested
-        collision_meshes = (
-            generate_collision_meshes(visual_mesh, method=collision_method, hull_count=hull_count)
-            if collision_method is not None
-            else []
-        )
-
-        # If the up axis is y, we need to rotate the meshes
-        if up_axis == "y":
-            rotation_matrix = trimesh.transformations.rotation_matrix(math.pi / 2, [1, 0, 0])
-            visual_mesh.apply_transform(rotation_matrix)
-            for collision_mesh in collision_meshes:
-                collision_mesh.apply_transform(rotation_matrix)
-
-        # If the scale is nonzero, we apply it to the meshes
-        if scale != 1.0:
-            scale_transform = trimesh.transformations.scale_matrix(scale)
-            visual_mesh.apply_transform(scale_transform)
-            for collision_mesh in collision_meshes:
-                collision_mesh.apply_transform(scale_transform)
-
-        # Check the bounding box size and complain if it's larger than 3 meters
-        bbox_size = visual_mesh.bounding_box.extents
-        click.echo(f"Visual mesh bounding box size: {bbox_size}")
-
-        if confirm_bbox:
-            if any(size > 3.0 for size in bbox_size):
-                click.echo(
-                    "Warning: The bounding box sounds a bit large. Are you sure you don't need to scale? "
-                    "We just wanted to confirm this is intentional. You can skip this check by passing --no-confirm-bbox."
-                )
-                click.confirm("Do you want to continue?", abort=True)
-
-            elif any(size < 0.01 for size in bbox_size):
-                click.echo(
-                    "Warning: The bounding box sounds a bit small. Are you sure you don't need to scale? "
-                    "We just wanted to confirm this is intentional. You can skip this check by passing --no-confirm-bbox."
-                )
-                click.confirm("Do you want to continue?", abort=True)
-
-        # Generate the URDF
-        click.echo(f"Generating URDF for {category}/{model}...")
-        generate_urdf_for_obj(visual_mesh, collision_meshes, category, model, overwrite=overwrite)
-        click.echo("URDF generation complete!")
-
-        urdf_path = None
-        collision_method = None
+        temp_urdf_dir = tempfile.mkdtemp()
+        temp_dirs.append(temp_urdf_dir)
+        try:
+            # Try to generate URDF, may raise ValueError if too many submeshes
```

Comment:
is this the only edge case we expect? if so, can we explicitly check for this condition vs. just a blind try / except?